### PR TITLE
[dompurify] enhance typings

### PR DESCRIPTION
**Pull request by falsandtru**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: no document, see sources https://github.com/cure53/DOMPurify/blob/master/src/purify.js

### CODE REVIEW

File: types/dompurify/index.d.ts [hunk 1]
```diff
@@ -3,32 +3,56 @@
...
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-interface IDOMPurify {
-  sanitize(s:string):string;
-  sanitize(s:string, config:IDOMPurifyConfig):any;
+declare module 'dompurify' {
+    export = DOMPurify;
+}
+
+declare var DOMPurify: DOMPurify;
```
Comment (DanielRosenwasser) [inline types/dompurify/index.d.ts]: I think this should be changed to be something like

```ts
declare var DOMPurify: IDOMPurify;
export = DOMPurify;
export as namespace DOMPurify;
```

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/dompurify/index.d.ts (hunk 1 above)
Comment (falsandtru) [inline types/dompurify/index.d.ts]: adopted.

File: types/dompurify/index.d.ts [hunk 2]
```diff
@@ -3,32 +3,63 @@
...
+    | 'uponSanitizeAttribute'
+    | 'afterSanitizeAttributes'
+    | 'beforeSanitizeShadowDOM'
+    | 'uponSanitizeShadowNode'
+    | 'afterSanitizeShadowDOM';
+
+type DOMPurifyHookEvent
+    = DOMPurifySanitizeElementHookEvent
+    | DOMPurifySanitizeAttributeHookEvent
+    | null;
```
Comment (falsandtru) [inline types/dompurify/index.d.ts]: Seems like TS cannot use nullable union types. Is this a spec?

Comment (DanielRosenwasser) [inline types/dompurify/index.d.ts]: What exactly do you mean by "nullable union types"?

Comment (falsandtru) [inline types/dompurify/index.d.ts]: I want to get `DOMPurifyHookEvent: DOMPurifySanitizeElementHookEvent | DOMPurifySanitizeAttributeHookEvent | null` but `null` is removed by compiler. So I actually got `DOMPurifyHookEvent: DOMPurifySanitizeElementHookEvent | DOMPurifySanitizeAttributeHookEvent`. This type doesn't have `null`. How can I add to this type possibility to be `null`?

Comment (DanielRosenwasser) [inline types/dompurify/index.d.ts]: I'm not sure; under `strictNullChecks` this should just work. Can you send a repro to the TS issue tracker?

Comment (falsandtru) [inline types/dompurify/index.d.ts]: I checked that type again then it works fine. Something was wrong. Now there is no problem. Please merge.

**Pull request merged**